app/models/verification.rb

```diff
@@ -0,0 +1,138 @@
+# frozen_string_literal: true
+
+# A Verification represents a user's confirmation that a tag is correctly
+# applied to an audio event.
+# @see (#AudioEvent) and (#Tag) for more information on these models.
+#
+# == Schema Information
+#
+# Table name: verifications
+#
+#  id             :bigint           not null, primary key
+#  confirmed      :enum             not null
+#  created_at     :datetime         not null
+#  updated_at     :datetime         not null
+#  audio_event_id :bigint           not null
+#  creator_id     :integer          not null
+#  tag_id         :bigint           not null
+#  updater_id     :integer
+#
+# Indexes
+#
+#  idx_on_audio_event_id_tag_id_creator_id_f944f25f20  (audio_event_id,tag_id,creator_id) UNIQUE
+#  index_verifications_on_audio_event_id               (audio_event_id)
+#  index_verifications_on_tag_id                       (tag_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (audio_event_id => audio_events.id) ON DELETE => cascade
+#  fk_rails_...  (creator_id => users.id)
+#  fk_rails_...  (tag_id => tags.id)
+#  fk_rails_...  (updater_id => users.id)
+#
+class Verification < ApplicationRecord
+  belongs_to :audio_event, inverse_of: :verifications
+  belongs_to :tag, inverse_of: :verifications
+  belongs_to :creator, class_name: 'User', inverse_of: :created_verifications
+  belongs_to :updater, class_name: 'User', inverse_of: :updated_verifications, optional: true
+
+  # A user can only have one verification per audio event and tag
+  validates :creator_id, uniqueness: { scope: [:audio_event_id, :tag_id] }
```

Comment:
if this is enforced a database constraint then this is doubling the work the database has to do.

it is not necessarily something to remove: it can be a nicer experience for rails validation errors to be reported and to also enforce integrity in the database, but we have to decide if it is worth the cost.

Reply:
Would that mean considering the front end implementation to understand the
utility here? Like, since adding a verification has to be unique
for the user / audio event / tag, choosing to verify something means that
information is set and the user can't manipulate them. So there is either an
existing verifcation for that combination or not, which changes the actions available to
show rather than create or something. In which case the rails verification
would be unnecessary?

Reply:
It's a good point that the validation won't be surfaced to the user of the website.

But it will be surfaced to consumers of the API. An error message with details is more useful than a generic 406 not acceptable.

Try it out! Remove the validation, write a request spec, see how ergonomic it is. Do we surface a useful error message? I can't remember!

Reply:
ActiveRecord::RecordNotUnique:
       PG::UniqueViolation: ERROR:  duplicate key value violates unique constraint "idx_on_audio_event_id_tag_id_creator_id_f944f25f20"
       DETAIL:  Key (audio_event_id, tag_id, creator_id)=(1, 1, 3) already exists.